packages/ncids-js/src/components/nci-header/utils/nci-focus-trap.ts

```diff
@@ -0,0 +1,156 @@
+/**
+ * FOCUS TRAP
+ *
+ *
+ * Initialize the FocusTrap component:
+ * ```
+ * FocusTrap.element.create(HTMLElement);
+ * ```
+ */
+
+export class FocusTrap {
+	/** Active Parent Component. */
+	protected element: HTMLElement;
+	/** The Element that is the contains. */
+	protected context: HTMLElement;
+	/** First focusable element */
+	protected firstFocusableElement: HTMLElement;
+	/** Last focusable element */
+	protected lastFocusableElement: HTMLElement;
+	/** List of element types to check */
+	protected focusableElements: string;
+	/** First focusable element */
+	protected focusableContent: NodeListOf<Element>;
+	/** Callback for handling the toggle functionality.  */
+	private eventListener: EventListener = (event) => this.checkTrap(event);
+	/** Map object of the component. */
+	private static _components: Map<HTMLElement, FocusTrap> = new Map<
+		HTMLElement,
+		FocusTrap
+	>();
+
+	/**
+	 * Sets component variables and initializes component.
+	 *
+	 * @param {HTMLElement} element Component being created.
+	 * @protected
+	 */
+	protected constructor(element: HTMLElement) {
+		this.element = element;
+		this.context = element;
+
+		this.focusableElements =
+			"button, [href], input, select, textarea, [tabindex]:not([tabindex='-1'])";
+		this.focusableContent =
+			this.element?.querySelectorAll(this.focusableElements) || [];
+		this.firstFocusableElement = <HTMLElement>(
+			this.element?.querySelectorAll(this.focusableElements)[0]
+		);
+		this.lastFocusableElement = <HTMLElement>(
+			this.focusableContent[this.focusableContent.length - 1]
+		);
+
+		const existingComponent = FocusTrap._components.get(this.element);
+		if (existingComponent) {
+			existingComponent.unregister();
+		}
+
+		this.initialize();
+		FocusTrap._components.set(this.element, this);
+	}
+
+	/**
+	 * Instantiates this component of the given element.
+	 *
+	 * @param {HTMLElement} element Component being created.
+	 * @public
+	 */
+	public static create(element: HTMLElement): FocusTrap {
+		if (!(element instanceof HTMLElement)) {
+			throw 'Element is not an HTMLElement';
+		}
+		return this._components.get(element) || new this(element);
+	}
+
+	/**
+	 * Resets component to a clean state.
+	 *
+	 * @public
+	 */
+	public unregister(): void {
+		FocusTrap._components.delete(this.element);
+	}
+
+	/**
+	 * Initializes the trap.
+	 *
+	 * @private
+	 */
+	private initialize(): void {
```

Comment:
Let's remove this for now since it's not doing anything. Can always add it back later if we need to.